crucible/src/Lang/Crucible/Simulator/RegValue.hs

```diff
@@ -361,3 +363,41 @@ muxVariant sym recf ctx = \p x y ->
                 p
                 (unVB (x Ctx.! i))
                 (unVB (y Ctx.! i))
+
+------------------------------------------------------------------------
+-- Equality
+
+-- | Equality of 'RegValue's.
+--
+-- This is only supported for a few types, see #1582.
+eqRegValue ::
+  forall sym tp.
+  IsInterpretedFloatExprBuilder sym =>
+  sym ->
+  TypeRepr tp ->
+  RegValue sym tp ->
+  RegValue sym tp ->
+  IO (Pred sym)
+eqRegValue sym tp x y =
+  case tp of
+    -- Base types
+    BoolRepr -> eqPred sym x y
+    BVRepr _width -> bvEq sym x y
+    ComplexRealRepr -> cplxEq sym x y
+    FloatRepr _ -> isEq sym x y
```

Comment:
Using `isEq` does work, but it might be worth using `floatEq` instead for symmetry with the other cases that use type-specific equality functions.